### PR TITLE
Handle more than one value for a url query parameter

### LINKED ISSUE #453: `query[queryDataKey].split is not a function` in search query transform
## Sentry link

https://openverse.sentry.io/share/issue/f7bcd3d050574f78b23936e923453e7c/

## Description

```
TypeError: query[queryDataKey].split is not a function
  File "server.js", line 12723, col 52, in <anonymous>
  File "server.js", in Array.forEach
  File "server.js", line 12709, col 15, in queryToFilterData
  File "server.js", line 4073, col 124, in Object.setSearchStateFromUrl
  File "app:///pinia.prod.cjs", line 341, col 30, in Object.<anonymous>
    ret = action.apply(this && this.$id === $id ? this : store, args);
  File "pages/search.js", line 1948, col 17, in asyncData
  File "server.js", in runMicrotasks
  File "node:internal/process/task_queues", line 96, col 5, in processTicksAndRejections
```

This stacktrace was a bit difficult to decode, but I was able to pin it down to this line:

https://github.com/WordPress/openverse-frontend/blob/94fb35469ad28cc3cfd3d1b9ef877b695925a736/src/utils/search-query-transform.ts#L202

It appears that, while `qquery[queryDataKey]` may not be falsy from the check above, it might not be the correct type which is expected of an object that has the `.split` function available.

## Reproduction

I was unable to reproduce this myself.

### LINKED ISSUE #1046: Frontend URL parsing: filterParameter.split is not a function
## Sentry link

https://openverse.sentry.io/share/issue/44140b9540954051b37d00f635382d4c/

## Description

We need to validate that `filterParameter` is a string and not an array of strings before trying to process it. If the parameter is an array, we should probably just use the first one because we join the values with a comma to pass 2 values (so, `license=by,cc0` instead of `license=by&license=cc0`).

## Reproduction

Go to a URL with 2 `sources` params: 
https://openverse.org/search/image?q=cat&source=thingiverse&source=images

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1046 by @obulat 
Fixes #453 by @AetherUnbound 

## Description

Vue router supports having separate values for a single parameter in the URL: `q=cat&q=dog`. This is parsed to `context.query` as `{ q: ["cat", "dog"] }`. Our code, however, expects the values for parameters to be strings, not arrays. This causes a 500 error in Nuxt for such URLs.

This PR pre-processes the `query` received from Nuxt context to leave only the first parameter if an array is given. We can safely discard the other values because normally we do not set such parameters using the UI. If we need to pass several values, we use a comma-separated value: `{ license: "by,sa" }` instead of an array. 

## Testing Instructions

Run the app and go to `localhost:8443/search/?q=cat&q=dog`. You should receive the results for `cat`, and `dog` should be discarded as a search term. If you do the same on main, you would get a Nuxt error screen.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/utils/search-query-transform.ts [hunk 1]
```diff
@@ -14,6 +14,9 @@ import {
 import { getParameterByName } from "~/utils/url-params"
 import { deepClone } from "~/utils/clone"
 
+import type { Context } from "@nuxt/types"
+import type { Dictionary } from "vue-router/types/router"
```
Comment (sarayourfriend) [inline frontend/src/utils/search-query-transform.ts:18]: Just a curiosity: what is the benefit of this type vs `Record<string, string>`?

**Review by sarayourfriend — COMMENTED** [review]
This looks fine to me, but I question whether this is the right approach (discarding all but the first item) for fields other than `q`. I understand why `q` should just be a single one as it's hard to assume what the intention is (should it be `AND` or `OR` or something else entirely), but for the other ones, why wouldn't we just combine them into a comma separated string instead?

I guess part of this would be answered if I knew more about the context of _when_ this happens to real users. What is the situation where this occurs? If we explicitly do not want to support this type of query, then I think the more expected approach would be to return a 400 Bad Request rather than trying to assume what the requester intended.

**Comment by obulat** [comment]
> I guess part of this would be answered if I knew more about the context of _when_ this happens to real users. What is the situation where this occurs? If we explicitly do not want to support this type of query, then I think the more expected approach would be to return a 400 Bad Request rather than trying to assume what the requester intended.

I think this happens when the users are constructing the URL manually. Moreover, this might be an attempt to hack something as the `q` in such requests can be unusual.

**Review by sarayourfriend — APPROVED** [review]
LGTM! I still think returning a 400 instead of still trying to process is the best solution for API usage we do not want to support (especially if we think it might be related to abusive usage, programmatic?), but treating the behaviour as "undefined" and doing whatever is easiest is also fine.

**Pull request merged**